SignalMessaging/environment/AppSetup.m

```diff
@@ -32,20 +28,39 @@ + (void)setupEnvironmentWithCallMessageHandlerBlock:(CallMessageHandlerBlock)cal
     static dispatch_once_t onceToken;
     dispatch_once(&onceToken, ^{
         // Order matters here.
+        //
+        // All of these "singletons" should have any dependencies used in their
+        // initializers injected.
         [[OWSBackgroundTaskManager sharedManager] observeNotifications];
 
-        [Environment setCurrent:[Release releaseEnvironment]];
+        OWSPrimaryStorage *primaryStorage = [[OWSPrimaryStorage alloc] initStorage];
+        [OWSPrimaryStorage protectFiles];
 
-        id<OWSCallMessageHandler> callMessageHandler = callMessageHandlerBlock();
-        id<NotificationsProtocol> notificationsManager = notificationsManagerBlock();
+        OWSPreferences *preferences = [OWSPreferences new];
 
-        TextSecureKitEnv *sharedEnv =
-            [[TextSecureKitEnv alloc] initWithCallMessageHandler:callMessageHandler
-                                                 contactsManager:[Environment current].contactsManager
-                                                   messageSender:[Environment current].messageSender
-                                            notificationsManager:notificationsManager
-                                                  profileManager:OWSProfileManager.sharedManager];
-        [TextSecureKitEnv setSharedEnv:sharedEnv];
+        TSNetworkManager *networkManager = [[TSNetworkManager alloc] initDefault];
+        OWSContactsManager *contactsManager = [[OWSContactsManager alloc] initWithPrimaryStorage:primaryStorage];
+        ContactsUpdater *contactsUpdater = [ContactsUpdater new];
+        OWSMessageSender *messageSender = [[OWSMessageSender alloc] initWithNetworkManager:networkManager
+                                                                            primaryStorage:primaryStorage
+                                                                           contactsManager:contactsManager];
```

Comment:
`AppSetup` now creates the Environment's singletons in a specific order, injecting their dependencies, then hangs them on the Environments.